shared_model/backend/protobuf/permissions.hpp

```diff
@@ -0,0 +1,64 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef IROHA_SHARED_MODEL_PROTO_PERMISSIONS_HPP
+#define IROHA_SHARED_MODEL_PROTO_PERMISSIONS_HPP
+
+#include "interfaces/permissions.hpp"
+
+#include <boost/optional.hpp>
+#include <string>
+#include <vector>
+#include "primitive.pb.h"
+
+namespace shared_model {
+  namespace proto {
+    namespace permissions {
+
+      /**
+       * @param perm protocol object for conversion
+       * @return sm object if conversion can be done
+       */
+      boost::optional<interface::permissions::Role> fromTransport(
+          iroha::protocol::RolePermission perm) noexcept;
+      /**
+       * @param sm object for conversion
+       * @return protobuf object
+       */
+      iroha::protocol::RolePermission toTransport(
+          interface::permissions::Role r);
+      /**
+       * @param sm object for conversion
+       * @return its string representation
+       */
+      std::string toString(interface::permissions::Role r);
+
+      /**
+       * @param perm protocol object for conversion
+       * @return sm object if conversion can be done
+       */
+      boost::optional<interface::permissions::Grantable> fromTransport(
+          iroha::protocol::GrantablePermission perm) noexcept;
+      /**
+       * @param sm object for conversion
+       * @return protobuf object
+       */
+      iroha::protocol::GrantablePermission toTransport(
+          interface::permissions::Grantable r);
+      /**
+       * @param sm object for conversion
+       * @return its string representation
+       */
+      std::string toString(interface::permissions::Grantable r);
+
+      std::vector<std::string> toString(
+          interface::PermissionSet<interface::permissions::Role> set);
```

Comment:
Pass by reference?

Also add empty line between two methods please.

It would also be nice to add comment, with string example